src/OrchardCore/OrchardCore.ContentManagement.Abstractions/ContentExtensions.cs

```diff
@@ -225,13 +225,13 @@ public static ContentElement Apply(this ContentElement contentElement, ContentEl
         if (contentElement.Data is not null)
         {
             contentElement.Data.Merge(JObject.FromObject(element.Data), _jsonMergeSettings);
+            contentElement.Elements.Clear();
```

Comment:
Ideally this would access the `_element` field so it doesn't create the backing array if it's null. This would mean to make it internal.

```suggestion
            contentElement?._elements.Clear();
```

Reply:
I moved this to only clear if something is merged into the data object. In that case `Elements` will be most likely already materialized.